docs/use-cases/sensor-module.md

```diff
@@ -0,0 +1,392 @@
+current_pm2_5 = current.Variables(1).Value()
+
+print(f"Current time {current.Time()}")
+print(f"Current pm10 {current_pm10}")
+print(f"Current pm2_5 {current_pm2_5}")
+```
+
+{{< /expand >}}
+
+{{< expand "An example of getting PM2.5 and PM10 readings over serial from sensor hardware" >}}
+This script is based on the [pms7003 module](https://app.viam.com/module/joyce/pms7003).
+Before creating the module, the following script could have been used to verify the code that gets the readings from the sensor.
+
+```python {class="line-numbers linkable-line-numbers"}
+# my-sensor-test.py
+from serial import Serial
+import time
+
+def main():
+
+    port = Serial('/dev/ttyAMA0', baudrate=9600)
+
+    def parse_data(data):
+
+        if len(data) < 2:
+            return {}
+        if data[0] == 0x42 and data[1] == 0x4d:
+            pm2_5_atm = (data[12] << 8) + data[13]
+            pm10_atm = (data[14] << 8) + data[15]
+            print(f'PM2.5 (atmospheric): {pm2_5_atm} µg/m3')
+            print(f'PM10 (atmospheric): {pm10_atm} µg/m3')
+
+            # Return a dictionary of the readings
+            return {
+                "pm2_5_atm": pm2_5_atm,
+                "pm10_atm": pm10_atm,
+            }
+
+        else:
+            print('Data does not start with the expected start bytes.')
+            return {}
+
+    while port.in_waiting == 0:
+        time.sleep(0.01)  # wait for 10 ms
+
+    data = port.read(port.in_waiting)
+    print(parse_data(data))
+
+if __name__ == "__main__":
+    main()
+```
+
+{{< /expand >}}
+
+{{< expand "Another example (this one is for a base, not a sensor)" >}}
+
+An example from the [custom robot dog base tutorial](/tutorials/custom/custom-base-dog/).
+When you run this with the robot dog hardware, you can tell it works because the robot dog takes a few steps forward.
+
+```python {class="line-numbers linkable-line-numbers"}
+# dog_test.py is for testing the connection
+import socket
+import time
+
+s = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
+s.connect(("PASTE DOG IP ADDRESS HERE", 5001))
+
+cmd = "CMD_MOVE_FORWARD#15"
+s.send(cmd.encode("utf-8"))
+time.sleep(7)
+cmd = "CMD_MOVE_STOP"
+s.send(cmd.encode("utf-8"))
+cmd = "CMD_RELAX"
+s.send(cmd.encode("utf-8"))
+```
+
+{{< /expand >}}
+
+Run your test script from your command line and make sure you are able to get readings from the sensor before proceeding.
+
+## Generate boilerplate module code
+
+You need some boilerplate files to make your module.
+To make this easier, use the Viam module generator:
+
+1. Follow the steps in the [readme](https://github.com/viam-labs/generator-viam-module/tree/main) to install the generator and run it.
+
+1. When prompted for a model triplet, use `<your organization public namespace>:<repo name>:<what you want to call your sensor model>`.
+   For example, `jessamy:weather:meteo_PM`.
+
+   - You can find your organization namespace by going to your organization settings in the [Viam app](https://app.viam.com).
```

Comment:
might need more explanation, idk what to do in organization settings. Or is that next bullet point?

Reply:
I thought about this, but since nowadays we're erring on the side of less verbosity, if someone can figure out how to write a module I trust them to find this with a gesture in the right direction. I think clicking the name of their org, and then clicking "settings", is intuitive enough if they know they're looking for organization settings.